projects/gloo/pkg/plugins/aws/ec2/ec2_utils.go

```
@@ -64,10 +66,13 @@ func GetInstancesFromDescription(desc *ec2.DescribeInstancesOutput) []*ec2.Insta
 
 // this filter function defines what gloo considers a valid EC2 instance
 func validInstance(instance *ec2.Instance) bool {
-	if instance.PublicIpAddress == nil {
```

fix a bug where only instances with public ips were shown

---

This function should probably be removed and we need to do these checks once we know if the upstream wants to use the public or the private ips, otherwise we could deref a nil pointer

---

could you also add some tests in `Context("convert to endpoint", func() {` in `projects/gloo/pkg/plugins/aws/ec2/ec2_test.go:83` that test various states of with/without public/private ips?

I think we'll get a nil pointer deref error until we check the existence of the public/private ips when creating the endpoint.

---

actually, it won't panic, but it will pass an empty string, per behavior of `aws.StringValue(ipAddr)`. We should return an error in `upstreamInstanceToEndpoint` instead

---

we were already checking and returning nil, I just added a warning statement and some tests